### PR TITLE
[WIP] Virt apiserver

### PR DESCRIPTION
**Pull request by stu-gott**
- [x] proposal
- [x] manifests
- [x] etcd backend
- [ ] tests updated
- [ ] console/spice ported from virt-api

### CODE REVIEW

**Comment by fabiand** [comment]
Pfeh - This looks good so far.

**Comment by mpolednik** [comment]
I believe this feature lacks a proposal. Could you create a doc explaining the scope, which services are affected and how new services would be developed when using the aggregated apiserver? The doc being part of this PR is fine.

**Comment by stu-gott** [comment]
@mpolednik You are correct! I accidentally created this PR a while back, and it's very much a work in progress currently. I'll definitely add a proposal/explanation for these changes before removing the "[WIP]"

File: manifests/virt-apiserver.yaml.in [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            - "-v"
+            - "9"
+        ports:
+          - containerPort: 8183
+            name: "virt-apiserver"
+            protocol: "TCP"
+        volumeMounts:
+          - mountPath: /var/run/virt-apiserver-cert
+            name: virt-apiserver-cert
+      - name: etcd
```
Comment (jhernand) [inline manifests/virt-apiserver.yaml.in]: What is the reason for having a separate instance of `etcd` can't the API server use the same instance than the rest of the cluster? If the administrator of the cluster configures it for high availability and replication of the master, then this separate instance of `etcd` won't benefit from that, if I understand correctly.

**Comment by fabiand** [comment]
Great work, due to the missing in-cluster storage for user API servers we decided to stick with CCRDs for the time being, thus this user API server can be closed.

Once Kubernetes gains in-cluster user API server storage, then we might want to revisit this topic again.

Thanks Stu.

**Pull request closed without merging**